src/fetcher/fetcher.mts

```diff
@@ -122,7 +99,7 @@ function getFetchParams(request: Request) {
   const query = getQuery(request.method, payload, request.queryParams);
   const body = getBody(request.method, payload);
   const headers = getHeaders(body, request.init?.headers);
-  const url = request.baseUrl + path + query;
+  const url = new URL(query, new URL(path, request.baseUrl)).toString();
```

Comment:
this is the actual fix in here ^^

Reply:
while I think this whole `getQuery` thing is pretty ugly, yours is still an good improvement. but hell, we probably should clean this whole module up eventually.